sklearn/cluster/_bisect_k_means.py

```diff
@@ -0,0 +1,727 @@
+        leaves_dict = {
+            -1: {"samples": np.ones(X.shape[0], dtype=bool), "error_or_size": None}
+        }
+
+        # ID of biggest center stored in centers_dict
+        biggest_id = -1
+
+        last_center_id = 0
+
+        for n_iter in range(self.n_clusters - 1):
+            picked_samples = leaves_dict[biggest_id]["samples"]
+
+            # Pick data and weights to bisect
+            picked_data = X[picked_samples]
+            picked_weights = sample_weight[picked_samples]
+
+            # Perform Bisection
+            _centers, _labels = self._bisect(picked_data, picked_weights, random_state)
+
+            if self.verbose:
+                print(f"Centroid Found: {_centers[0]}")
+                print(f"Centroid Found: {_centers[1]}")
+
+            if strategy == "biggest_sse":
+                # "biggest_sse" uses computed SSE (Sum of Squared Errors)
+                # to pick cluster with biggest SSE Error
+                metrics_values = self._compute_bisect_errors(
+                    picked_data, _centers, _labels, picked_weights
+                )
+            else:
+                # strategy == 'largest_cluster'
+                # 'largest_cluster' takes counts occurances of each label
+                # to pick cluster with largest number of data points assigned
+                metrics_values = np.bincount(_labels)
+
+            # "Create Hierarchy":
+            # Cluster with smaller metrics value (SSE or ammount of points)
+            # will be at 'left side' and cluster with higher at 'right side'
+            lower_id = 0 if metrics_values[0] <= metrics_values[1] else 1
+            centers_id = [lower_id, 1 - lower_id]
+
+            # Assign calculated nested clusters to their root
+            tree_dict[biggest_id]["children"] = [last_center_id, last_center_id + 1]
+
+            for id in range(2):
+                child_id = tree_dict[biggest_id]["children"][id]
+                picked_id = centers_id[id]
+
+                # Save Results on Tree
+                tree_dict[child_id] = {"children": None, "center": _centers[picked_id]}
+
+                # Create Mask for samples for selecting proper data points
+                samples_mask = leaves_dict[biggest_id]["samples"].copy()
+                samples_mask[picked_samples] = _labels == picked_id
+
+                # Save recently generated leaves
+                leaves_dict[child_id] = {
+                    "samples": samples_mask,
+                    "error_or_size": metrics_values[picked_id],
+                }
+
+            # Split cluster is no longer leaf
+            del leaves_dict[biggest_id]
+
+            last_center_id += 2
+
+            # Pick new 'biggest cluster' to split
+            biggest_id, _ = max(
+                leaves_dict.items(), key=lambda x: x[1]["error_or_size"]
+            )
+
+        # Delete Initial cluster
+        del tree_dict[-1]
+
+        # Sort Leaf centers by their root
```

Comment:
Why do you sort leaves here ?

Reply:
Wrong understanding of how centers should be shown to see hierarchy.
I will remove that.